Source/Widgets/Geocoder/GeocoderViewModel.js

```diff
@@ -53,33 +66,183 @@ define([
         }
         //>>includeEnd('debug');
 
+        this._geocoderServices = options.geocoderServices;
+        if (!defined(options.geocoderServices)) {
+            this._geocoderServices = [
+                new LongLatGeocoderService(),
+                new BingMapsGeocoderService(),
+                new OpenStreetMapNominatimGeocoderService()
+            ];
+        }
+
+        var errorCredit;
         this._url = defaultValue(options.url, 'https://dev.virtualearth.net/');
         if (this._url.length > 0 && this._url[this._url.length - 1] !== '/') {
             this._url += '/';
         }
 
         this._key = BingMapsApi.getKey(options.key);
-        var errorCredit = BingMapsApi.getErrorCredit(options.key);
+        this._defaultGeocoderOptions = {
+            url: this._url,
+            key: this._key
+        };
+
+        if (defined(options.key)) {
+            errorCredit = BingMapsApi.getErrorCredit(options.key);
+        }
         if (defined(errorCredit)) {
             options.scene._frameState.creditDisplay.addDefaultCredit(errorCredit);
         }
 
+        this._viewContainer = options.container;
         this._scene = options.scene;
         this._flightDuration = options.flightDuration;
         this._searchText = '';
         this._isSearchInProgress = false;
         this._geocodeInProgress = undefined;
         this._complete = new Event();
+        this._suggestions = knockout.observableArray();
+        this._selectedSuggestion = knockout.observable();
+        this._showSuggestions = knockout.observable(true);
+        this._updateCamera = updateCamera;
 
         var that = this;
+
+        this._suggestionsVisible = knockout.pureComputed(function () {
+            return that._suggestions().length > 0 && that._showSuggestions();
+        });
+
         this._searchCommand = createCommand(function() {
+            if (defined(that._selectedSuggestion())) {
+                that.activateSuggestion(that._selectedSuggestion());
+                return false;
+            }
             if (that.isSearchInProgress) {
                 cancelGeocode(that);
             } else {
-                geocode(that);
+                geocode(that, that._geocoderServices);
             }
         });
 
+        this._adjustSuggestionsScroll = function (focusedItemIndex) {
```

Comment:
Is this used anywhere by the view?  If not, make this a private function `function adjustSuggestionsScroll`, move it above the constructor and pass in whatever values you need as params